cartography/intel/aws/identitystore.py

```diff
@@ -210,41 +212,59 @@ def get_list_account_assignments(boto3_session: boto3.session.Session, instance_
     return assignments
 
 
+@timeit
+def get_list_account_assignments_for_principal(boto3_session: boto3.session.Session, instance_arn: str, PrincipalId: str, PrincipalType: str, region: str):
+    client = get_boto3_client(boto3_session, 'sso-admin', region)
```

Comment:
let's not create the client for each user and group. have a common method outside and create the client there and re-use the client.